backend/src/main/java/ba/unsa/etf/si/bbqms/admin_service/implementation/DefaultStationService.java

```diff
@@ -128,4 +128,14 @@ public Set<TellerStation> getAllOfferingService(final Service service) {
     public Optional<TellerStation> findById(final long serviceId) {
         return this.tellerStationRepository.findById(serviceId);
     }
+
+    @Override
+    public Set<Service> getServicesForTellerStation(long stationId) {
+        TellerStation tellerStation = tellerStationRepository.findById(stationId).orElse(null);
+        if (tellerStation != null) {
+            return tellerStation.getServices();
+        } else {
+            throw new RuntimeException("Teller station not found");
+        }
+    }
```

Comment:
This can be simplified:
```
tellerStationRepository.findById(stationId)
                .map(TellerStation::getServices)
                .orElseThrow(EntityNotFoundException::new);
```